### PR TITLE
Unit testing: adds test_algebraic.py and more tests for MatricesGtest.cpp

### PR DESCRIPTION
**Pull request by PetholzA**
this PR splits up #1062 into semantically linked parts

### CODE REVIEW

File: networkit/test/test_algebraic.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		self.assertEqual(A[3][3], 0.0)
+		self.assertEqual(A[4][4], 0.0)
+		A2 = nk.algebraic.adjacencyMatrix(G, "sparse")
+		self.assertIsInstance(A2, scipy.sparse.csr_matrix)
+		B = nk.algebraic.laplacianMatrix(G)
+		self.assertIsInstance(B, scipy.sparse.coo_matrix)
+		eigenB = nk.algebraic.laplacianEigenvectors(G)
+		self.assertIsInstance(eigenB[0][0], np.complex128)		
+
+	def testEigenvectorValues(self):
```
Comment (angriman) [inline networkit/test/test_algebraic.py:26]: Here you are just testing that the first eigenvector type is complex. Shouldn't you check that the function produces the expected eigenvectors?

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 2]
```diff
@@ -1200,6 +1212,11 @@ TEST_F(MatricesGTest, testDimension) {
     testDimension<DenseMatrix>();
 }
 
+TEST_F(MatricesGTest, testMatrixToGraph) {
+    testMatrixToGraph<DynamicMatrix>();
+    testMatrixToGraph<CSRMatrix>();
+}
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:1218]: There is a more elegant way to instantiate a type-parametrized test suites: http://google.github.io/googletest/reference/testing.html#TYPED_TEST_SUITE_P

Anyway, this is not something introduced by this PR. This is ok for now, but we should change it later.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 3]
```diff
@@ -120,11 +120,23 @@ class MatricesGTest : public testing::Test {
...
     // TODO: Test mmT multiplication, etc.!
 };
 
+template <class Matrix>
+void MatricesGTest::testMatrixToGraph() {
+
+    Matrix mat = Matrix::adjacencyMatrix(METISGraphReader{}.read("input/power.graph"));
+    Graph G = MatrixTools::matrixToGraph(mat);
+
+    EXPECT_EQ(G.numberOfEdges(), 6594);
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:137]: What about testing also the number of nodes and that the produced edges are correct? Further, which kind of graph do we expect? Should be weighted/unweighted, directed/undirected? I think these properties should be tested.

**Review by angriman — CHANGES_REQUESTED** [review]
Thank you very much for the new tests. I left some comments.

**Comment by fabratu** [comment]
Currently, the coverage is at 81%. Since the module is small, try to reach 100%.

**Review by angriman — APPROVED** [review]

**Pull request merged**